tests/test_jsonschema.py

```diff
@@ -567,6 +567,53 @@ def test_duplicate_type_and_type_references_are_extracted_to_definitions(self):
         assert result['properties']['foo'] == {'$ref': '#/definitions/MyString'}
         assert result['properties']['bar'] == {'$ref': '#/definitions/MyString'}
 
+    def test_duplicate_types_multiple_calls_do_not_omit_load_only(self):
```

Comment:
Suggest to add multiple tests, which for some reason are missing from the beginning, that checks that the conflict resolving algorithm still works correctly for different modes because existing tests cover cases which don't use any modes. What should be checked is that the mode suffix doesn't break the conflict resolving.

Reply:
Extended `test_definition_name_conflict_resolving` and `test_unnamed_types_definition_name_conflict_resolving`